### PR TITLE
Add subjects and units to side nav

### PR DESCRIPTION
**Pull request by blakes24**
Updates sidebar to include subjects with icons and unit submenus

Screen Shot 2021-04-23 at 10 35 07 AM

### CODE REVIEW

File: components/layout/sidebar.js [hunk 1]
```diff
@@ -71,18 +86,36 @@ export default function SideBar() {
...
           </MenuItem>
-          <SubMenu
-            title="Components"
-            icon={<FontAwesomeIcon icon={faHeart} className="cursor-pointer" />}
-          >
-            <MenuItem>Component 1</MenuItem>
-            <MenuItem>Component 2</MenuItem>
-          </SubMenu>
+          {subjects &&
+            subjects.map((s) => (
```
Comment (dgriego) [inline components/layout/sidebar.js]: we should avoid single name variables.  I know you used them in a previous PR, but it was all on one line so I didn't think it was a big deal.  But there is no cost to writing out the variable names and is much more readable.
```
subjects.map((subject) => (
  title={`${subject.name} - ${subject.grade}`}
```

same for 
```
subject.units.map((unit) => (
```

Comment (dgriego) [reply components/layout/sidebar.js]: also, generally, it's rarely a good practice to use single variable names when working on a team, unless there are exceptions made or the team has decided in particular cases that single variable names don't matter in certain contexts (like the one I mentioned above)

File: components/layout/sidebar.js [hunk 2]
```diff
@@ -71,18 +86,36 @@ export default function SideBar() {
           className="border-t-2 border-white bg-blue"
           popperArrow
         >
-          <MenuItem
-            icon={<FontAwesomeIcon icon={faGem} className="cursor-pointer" />}
-          >
-            Dashboard
+          <MenuItem>
```
Comment (dgriego) [inline components/layout/sidebar.js]: we should add an icon for Dashboard here, I don't think Tara knew we could have icons when the sidebar was minimized.  Just use `faHome` for now, it just looks odd to have `D...` when it's collapsed

**Review by dgriego — CHANGES_REQUESTED** [review]
Looks good, nice work, just a couple of comments.  I was curious if you looked into an active color for the unit link if the unit was clicked or it matched the URL?  No need to implement for this iteration, but was curious.

File: components/layout/sidebar.js (hunk 1 above)
Comment (blakes24) [reply components/layout/sidebar.js]: ok, will do

File: components/layout/sidebar.js (hunk 2 above)
Comment (blakes24) [reply components/layout/sidebar.js]: Yeah, I had an icon for the dashboard at first then noticed it wasn't in the design. I agree it looks betting in the collapsed view to have one. I'll put it back in.

**Comment by blakes24** [comment]
> I was curious if you looked into an active color for the unit link if the unit was clicked or it matched the URL? No need to implement for this iteration, but was curious.

I was working on setting an active color for the link to the current page but I was having trouble overriding the styles from the react-pro-sidebar so I figured I'd get this up then look into it more.

**Pull request merged**